eth-providers/src/base-provider.ts

```diff
@@ -536,24 +534,25 @@ export abstract class BaseProvider extends AbstractProvider {
       ? (await this.getEvmAddress(headerExtended.author.toString())).toLowerCase()
       : DUMMY_ADDRESS;
 
-    const evmExtrinsicIndexes = getEvmExtrinsicIndexes(blockEvents);
+    let receipts: FullReceipt[];
+    if (receiptsFromSubql?.length) {
+      receipts = receiptsFromSubql.map(subqlReceiptAdapter);
+    } else {
+      /* ----------
+         if nothing is returned from subql, either no tx exists in this block,
+         or the block not finalized. So we still need to ask block cache.
+                                                                    ---------- */
+      receipts = await isCanonical
+        ? this.blockCache.getAllReceiptsAtBlock(blockHash)
+        : [];
```

Comment:
```suggestion
      receipts = this.blockCache.getAllReceiptsAtBlock(blockHash);
```
if `_blockTag` is a forked block hash,  we should return the actual receipts of the block

Reply:
at first I implemented that way, https://github.com/AcalaNetwork/bodhi.js/pull/662/commits/ccc11ff6a2bb550cea74879bcda16f92ab8df6a7#diff-8e05d01e1b336d3501a54605915fd1c9116c51bc23417352c54359c4ca197119L544-L548

but after a second thought, that might cause some inconsistency that when use call getblock, they saw transactions in it, but when they call get receipts for these tx, it will return `null`. What do you think?

Reply:
This looks insistent to me
```
const block = getBlock(forked-block-hash); 
const firstTx = block.tx[0];
const receipt = getReceipt(firstTx.hash);  // null
```